### DISCUSSION TITLE
pip install microstructpy issues

### DISCUSSION THREAD

**Discussion by dpd5121** [kip-hart/MicroStructPy#78] [Q&A] [answered]
I am running windows 10, python 3.10.8, and pip-22.3 . I have tried multiple versions of python but cannot get microstructpy to install properly. Most recently I ran the following based on the two pages below: pip install matplotlib - installed properly. pip install pybind11 - ran fine. Images of the current error are attached. Any help is appreciated. 
https://github.com/kip-hart/MicroStructPy/issues/19
https://docs.microstructpy.org/en/latest/
![image](https://user-images.githubusercontent.com/116897491/198641071-7bb73ce7-b98a-4ca1-94e0-15c4ea17d918.png)
![image](https://user-images.githubusercontent.com/116897491/198641803-cfabbfbf-274f-4659-8ec2-e094168e2c6b.png)

**Comment by kip-hart** [comment]
Hi @dpd5121 I'm sorry you're having trouble installing microstructpy. It looks like there's a snag with pyvoro. I'll take a look at it this weekend and let you know what I find.

Thanks for including your OS and Python version. It looks like the root cause is version/deprecation related.

**Reply by kip-hart** [reply]
Linking this CI job, windows 10 py310, for reference 

https://github.com/kip-hart/MicroStructPy/actions/runs/3016596917/jobs/4849465036

**Reply by kip-hart** [reply]
Hi @dpd5121 could you provide more of the output? Feel free to copy/paste the text directly into GitHub, rather than screenshots. Specifically, I'd also like to see the commands that created the error so I can reproduce it.

Another thing is the error message says "error: microsoft visual c++ 14.0 or greater is required". This could be the root of the problem. Could you check on that?

**Reply by kip-hart** [reply]
Upgrading setuptools may also help https://stackoverflow.com/questions/73257839/setup-py-install-is-deprecated-warning-shows-up-every-time-i-open-a-terminal-i

**Reply by dpd5121** [reply]
Thank you. Visual C++ appeared to be the issue. I must have missed that. I was able to successfully install microstructpy now. Thank you.

**Reply by mkaradge** [reply]
I similarly got stuck with installing microstructpy today afternoon. Thank you for this thread; I was able to install after updating Visual C++.

**Reply by kip-hart** [reply]
I'm glad you were able to resolve the issue. I'll update the troubleshooting guide with an entry on Visual Studio.

If you have any other questions or feedback while using MicrostructPy, please let me know. I'm open to ideas and suggestions for features.

**Comment by kip-hart** [comment, marked as answer]
To summarize the solution from the thread earlier, upgrading Visual C++ resolved this issue.

**Reply by dpd5121** [reply]
Also, it appears that you only need to install the Desktop Development with C++ Workload in Microsoft C++ Build Tools - Visual Studio to resolve the issue. (https://visualstudio.microsoft.com/visual-cpp-build-tools/)
